src/routes/oath.ts

```diff
@@ -162,4 +164,66 @@ router.get("/hubspot_refresh_token", async (req, res) => {
 	}
 });
 
+
+export const HubApiCall:any = async function (myapifunc:Function,refresh_token:string)
+{
+
+	// first try to see if the api call goes through if it does then send response back
+	try
+	{
+		let result = await myapifunc()
+		// api call went through everything is fine
+		return result;
+	}
+	catch(e)
+	{
+		try
+		{
+			//error in the api call get a new access token
+			let result = await getHubspotAccessToken(refresh_token);
+
+			if(result.response.status == 400)
```

Comment:
@davidbishoppp  If the getHubSpotAccessToken returns error it is handled here. JSON.parse(result) on line 195 returns a promise rejection that will be caught in the callee.

Reply:
Line 215 will is the scenario where the getHubspotAccessToken rejects the promise.